### ISSUE TITLE
Notifier never gets called

### ISSUE THREAD

**Issue by ignatz** [Renesanse/worker_manager#80]
The test here:

https://github.com/Renesanse/worker_manager/blob/master/test/update_task_completion_progress_test.dart#L34

is written in a way that it also passes when nothing is received, i.e. the listener never gets called, which is the case.

**Comment by Renesanse** [comment]
@ignatz thx for founding a bug, can you reproduce it when fake flag is false?

**Comment by ignatz** [comment]
Yes. You can verify by fixing the test first:

```dart
    test('onUpdateProgress should get value from sendPort', () async {
      final totalTitle = 'Total current: ';

      final values = <int>[];
      final result = await executor.execute(
          arg1: totalTitle,
          fun1: isolateTask,
          fake: false,
          notification: (int value) {
            print('onUpdateProgress: $value');
            values.add(value);
          });

      expect(values, equals([1, 2, 3, 4, 5]));

      expect(result, 15);
    });
```
will yield:

```
Progress should get value from sendPort [E]
  Expected: [1, 2, 3, 4, 5]
    Actual: []
     Which: at location [0] is [] which shorter than expected

  package:test_api                                     expect
  test/update_task_completion_progress_test.dart 34:7  main.<fn>.<fn>
```

Previously, the test was written as

```
stream.listen((event) {
  print('stream::listen:Event: $event');
  expect([1, 2, 3, 4, 5], containsValue(event));
});
```

which asserts nothing.

**Comment by Renesanse** [comment]
thx for opening the issue, resolved in last commit, pls try 5.0.2 version from pub.dev
